repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,193 @@
+
+        If the current repair is not yet assigned to a group, a new
+        `repair.order.group` is automatically created first.
+        The new repair will reuse basic logistics fields such as
+        partner, company, and locations, but not copy any parts or products.
+        """
+        self.ensure_one()
+        if not self.group_id:
+            self.group_id = self.env["repair.order.group"].create({})
+
+        # Create new empty repair record within the same group
+        new_repair = self.env["repair.order"].create(
+            {
+                "group_id": self.group_id.id,
+                "partner_id": self.partner_id.id,
+                "company_id": self.company_id.id,
+                "picking_type_id": self.picking_type_id.id,
+                "location_id": self.location_id.id,
+                "location_dest_id": self.location_dest_id.id,
+            }
+        )
+
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "repair.order",
+            "res_id": new_repair.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+
+    def write(self, vals):
+        """Propagate partner changes across all repairs in the same group.
+
+        When a user changes the partner on one repair, all other repairs
+        of the same group automatically receive the same partner, ensuring
+        data consistency. Recursion is prevented via the context flag
+        `skip_group_sync`.
+        """
+        res = super().write(vals)
+        if "partner_id" in vals and not self.env.context.get("skip_group_sync"):
+            for rec in self:
+                if rec.group_id:
+                    sib = rec.grouped_repair_ids
+                    if sib:
+                        sib.with_context(skip_group_sync=True).write(
+                            {"partner_id": vals["partner_id"]}
+                        )
+        return res
+
+    def _action_repair_confirm(self):
+        """Confirm repair and cascade confirmation to other group members."""
+        res = super()._action_repair_confirm()
+        if not self.env.context.get("skip_group_confirm"):
+            for rec in self:
+                sib = rec.grouped_repair_ids.filtered(lambda r: r.state == "draft")
+                if sib:
+                    sib.with_context(skip_group_confirm=True)._action_repair_confirm()
+        return res
+
+    def action_repair_cancel(self):
+        """Cancel repair and cascade cancellation to other group members."""
+        res = super().action_repair_cancel()
+        if not self.env.context.get("skip_group_cancel"):
+            for rec in self:
+                sib = rec.grouped_repair_ids.filtered(
+                    lambda r: r.state not in ("cancel", "done")
+                )
+                if sib:
+                    sib.with_context(skip_group_cancel=True).action_repair_cancel()
+        return res
+
+    def action_create_sale_order(self):
```

Comment:
This function is not working properly because you didn't check the original function while working on your code.